pyccel/ast/numpyext.py

```diff
@@ -1960,19 +1960,26 @@ def is_elemental(self):
 
 class NumpyAmax(NumpyUfuncUnary):
     """Represent a call to the amax function in the Numpy library"""
-    __slots__ = ()
+    __slots__ = ('_dtype','_precision','_class_type')
     name = 'amax'
-    def _set_shape_rank(self, x):
-        self._shape     = None
-        self._rank      = 0
+    _rank  = 0
+    _shape = None
+    _order = None
 
-    def _set_dtype_precision(self, x):
-        self._dtype     = x.dtype
-        self._precision = get_final_precision(x)
+    def __init__(self, arg):
+        if not isinstance(arg, TypedAstNode):
+            raise TypeError('Unknown type of  %s.' % type(arg))
+        super().__init__(arg)
+        if isinstance(arg.dtype, NativeBool):
+            self._dtype = NativeInteger()
+        else:
+            self._dtype = arg.dtype
+        self._class_type = self._dtype
+        self._precision = max(arg.precision, default_precision[self._dtype])
 
     @property
-    def is_elemental(self):
-        return False
```

Comment:
Why is this function removed?

Reply:
the function was returning false so i removed it, now i have added it again and it returns true since numpy.max() is an elemental function...tell me if i am wrong and need to change it

Reply:
I don't think that max is an elemental function. An elemental function is one where the function is applied elementwise. I.e

f([x, y, z])

is equivalent to:

[f(x), f(y), f(z)]

I don't think this is the case for `max`